src/test/java/io/vertx/test/core/SchedulerInterceptorTest.java

```diff
@@ -0,0 +1,167 @@
+    Context ctx = vertx.getOrCreateContext();
+
+    int interceptorNum = 10;
+    AtomicInteger cnt = new AtomicInteger();
+
+    for (int i = 0; i < interceptorNum; i++) {
+      final int expectedCount = i;
+      vertx.addSchedulerInterceptor((c, r) -> {
+        if (c == ctx) {
+          assertEquals(expectedCount, cnt.getAndIncrement());
+        }
+        return r;
+      });
+    }
+
+    ctx.runOnContext(v -> {
+      assertEquals(interceptorNum, cnt.get());
+      testComplete();
+    });
+
+    await();
+  }
+
+  @Test
+  public void testExecuteBlocking() {
+    Context ctx = vertx.getOrCreateContext();
+    AtomicInteger cnt = new AtomicInteger();
+
+    vertx.addSchedulerInterceptor((c, r) -> {
+      if (c == ctx) {
+        cnt.incrementAndGet();
+      }
+      return r;
+    });
+
+    ctx.executeBlocking(f -> {
+      assertEquals(1, cnt.get());
+    }, true, null);
+
+    ctx.executeBlocking(f -> {
+      assertEquals(2, cnt.get());
+      testComplete();
+    }, true, null);
+
+    await();
+  }
+
+  @Test
+  public void testTimer() {
+    Context ctx = vertx.getOrCreateContext();
+    AtomicInteger cnt = new AtomicInteger();
+
+    vertx.addSchedulerInterceptor((c, r) -> {
+      if (c == ctx) {
+        cnt.incrementAndGet();
+      }
+      return r;
+    });
+
+    ctx.runOnContext(v -> {
+      assertEquals(1, cnt.get());
+      vertx.setTimer(1000, h -> {
+        assertEquals(2, cnt.get());
+        vertx.runOnContext(v2 -> {
+          assertEquals(3, cnt.get());
+          testComplete();
+        });
+      });
+    });
+
+    await();
+  }
+
+  @Test
+  public void testRunPeriodic() {
+    Context ctx = vertx.getOrCreateContext();
+    AtomicInteger cnt = new AtomicInteger();
+
+    vertx.addSchedulerInterceptor((c, r) -> {
+      if (c == ctx) {
+        cnt.incrementAndGet();
+      }
+      return r;
+    });
+
+    ctx.runOnContext(v -> {
+      assertEquals(1, cnt.get());
+      vertx.setPeriodic(1000, h -> {
+        assertEquals(2, cnt.get());
+        vertx.cancelTimer(h);
+        vertx.runOnContext(v2 -> {
+          assertEquals(3, cnt.get());
+          testComplete();
+        });
+      });
+    });
+
+    await();
+  }
+
+  @Test
+  public void testRemoveInterceptor() {
+    AtomicInteger cnt1 = new AtomicInteger();
+    AtomicInteger cnt2 = new AtomicInteger();
+
+    BiFunction<Context, Runnable, Runnable> si1 = (c, r) -> {
+      cnt1.incrementAndGet();
+      return r;
+    };
+
+    BiFunction<Context, Runnable, Runnable> si2 = (c, r) -> {
+      cnt2.incrementAndGet();
+      return r;
+    };
+
+    vertx.addSchedulerInterceptor(si1);
+    vertx.addSchedulerInterceptor(si2);
+
+    vertx.runOnContext(v -> {
+      assertEquals(1, cnt1.get());
+      assertEquals(1, cnt2.get());
+      vertx.removeSchedulerInterceptor(si2);
+    });
+    vertx.runOnContext(v -> {
+      assertEquals(2, cnt1.get());
+      assertEquals(1, cnt2.get());
+      testComplete();
+    });
+
+    await();
+  }
```

Comment:
Once the event bus interceptors are merged, we should also have a test between 2 verticles, and 1 regular verticle and 1 worker verticle.